### DISCUSSION TITLE
Global volatile variables

### DISCUSSION THREAD

**Discussion by ethanrjs** [SkriptLang/Skript#4878] [Suggestions] [answered]
I think it would be incredibly useful if Skript had variables that were automatically deleted upon the server stopping (i.e. stored in memory and never written to a file)

They would be different from local variables in that they could be accessed globally but similar in that they're stored purely in memory.

Here's an example of what it could look like:
```
on chat:
    set {var} to message
    set {!var} to message
    set {_var} to message

command /showMsg:
    trigger:
        broadcast {var} # "hi"
        broadcast {!var} # "hi"
        broadcast {_var} " <none>
on load:
    broadcast {var} # "hi"
    broadcast {!var} # <none>
    broadcast {_var} # <none>
```

With this, it could add a more resource efficient method of storing and retrieving temporary values, as well as create solutions for things like cooldowns and temporary variables that need to be accessed in a global scope.

Although storing items in a list variable and clearing the entire list is a way to manage these variables, it would be more resource efficient and would give some extra syntactic sugar to using these variables in this manner.

**Comment by sovdeeth** [comment, marked as answer]
This already exists. Check the last section of the config.sk file, there's an explanation on how to enable this there.

> PS: If you don't want some variables to be saved in any database (e.g. variables that contain an %entity% which usually despawn when the server is shut down)
> you can modify the last database's pattern to not match all variables, e.g. use '(?!x_).\*' to match all variables that don't start with 'x_'.
> Be very cautious when doing this however as unsaved variables cannot be recovered after the server has been stopped.
> I recommend to use a single character to denote unsaved variables (similar to local variables' '_'), e.g. '-', in which case the last database's pattern should be '(?!-).\*'.
